packages/grid/x-data-grid-pro/src/components/headerFiltering/GridHeaderFilterItem.tsx

```diff
@@ -0,0 +1,298 @@
+            }
+            apiRef.current.startHeaderFilterEditMode(colDef.field);
+            break;
+          case 'Tab': {
+            if (isEditing) {
+              const fieldToFocus = columnFields[colIndex + (event.shiftKey ? -1 : 1)] ?? null;
+
+              if (fieldToFocus) {
+                apiRef.current.startHeaderFilterEditMode(fieldToFocus);
+                apiRef.current.setColumnHeaderFilterFocus(fieldToFocus, event);
+              }
+            }
+            break;
+          }
+          default:
+            if (isEditing || event.metaKey || event.ctrlKey || event.altKey || event.shiftKey) {
+              break;
+            }
+            apiRef.current.startHeaderFilterEditMode(colDef.field);
+            break;
+        }
+      },
+      [apiRef, colDef.field, colIndex, columnFields, headerFilterMenuRef, isEditing, isMenuOpen],
+    );
+
+    const publish = React.useCallback(
+      (eventName: keyof GridHeaderFilterEventLookup, propHandler?: React.EventHandler<any>) =>
+        (event: React.SyntheticEvent) => {
+          apiRef.current.publishEvent(
+            eventName,
+            apiRef.current.getColumnHeaderParams(colDef.field),
+            event as any,
+          );
+          if (propHandler) {
+            propHandler(event);
+          }
+        },
+      [apiRef, colDef.field],
+    );
+
+    const onClick = React.useCallback(
+      (event: React.MouseEvent) => {
+        if (!hasFocus) {
+          apiRef.current.startHeaderFilterEditMode(colDef.field);
+          apiRef.current.setColumnHeaderFilterFocus(colDef.field, event);
+        }
+      },
+      [apiRef, colDef.field, hasFocus],
+    );
+
+    const mouseEventsHandlers = React.useMemo(
+      () => ({
+        onKeyDown: publish('headerFilterKeyDown', onKeyDown),
+        onClick: publish('headerFilterClick', onClick),
+      }),
+      [onClick, onKeyDown, publish],
+    );
+
+    const ownerState = {
+      ...rootProps,
+      colDef,
+    };
+
+    const classes = useUtilityClasses(ownerState as OwnerState);
+
+    const isNoInputOperator = NO_INPUT_OPERATORS[colDef.type!]?.includes(item.operator);
+    const isFilterActive = hasFocus || Boolean(item?.value) || isNoInputOperator;
+    const label =
+      OPERATOR_LABEL_MAPPING[item.operator] ??
+      apiRef.current.getLocaleText(
+        `filterOperator${capitalize(item.operator)}` as 'filterOperatorContains',
+      );
+
+    return (
+      <div
+        className={clsx(classes.root, headerClassName)}
+        ref={handleRef}
+        style={{
+          height,
+          width,
+          minWidth: width,
+          maxWidth: width,
+        }}
+        role="columnheader"
+        aria-colindex={colIndex + 1}
+        aria-label={headerFilterComponent == null ? colDef.headerName ?? colDef.field : undefined}
+        {...other}
+        {...mouseEventsHandlers}
+      >
+        {headerFilterComponent}
+        {InputComponent && !headerFilterComponent ? (
```

Comment:
It looks like `renderHeaderFilter: () => null` doesn't work as expected and renders the default component:
https://codesandbox.io/s/boring-colden-m260of?file=/demo.tsx

Reply:
Fixed, it should be working now
https://codesandbox.io/s/aged-voice-jnvow4?file=/demo.tsx